internal/node/node.bzl

```diff
@@ -139,14 +139,21 @@ def _to_manifest_path(ctx, file):
         return ctx.workspace_name + "/" + file.short_path
 
 def _to_execroot_path(ctx, file):
-    parts = file.path.split("/")
+    path = file.path
+
+    # most likely a TreeArtifact exported by js_library with exports directories only.
+    # since a TreeArtifact is generated by an action, it resides inside bazel-out directory.
+    if file.is_directory and file.dirname.startswith(ctx.bin_dir.path + "/"):
+        path = path.replace(ctx.bin_dir.path + "/", "")
```

Comment:
oh. you had to drop the bazel-out portion here? why was that?

Reply:
when the file is a tree artifact, it is going to be created by an action thus bazel will put it into the bazel-out directory. something like  `bazel-out/darwin_arm64-fastbuild/bin/external/npm_directory_artifacts/node_modules/pkg_with_bin`  
then linker will pick up this package and put it into `execroot/node_modules`.
so in order to perform the mapping to the linked location, it has to get rid of the bin_dir portion.

`bazel-out/darwin_arm64-fastbuild/bin/external/npm_directory_artifacts/node_modules/pkg_with_bin` -> `node_modules/pkg_with_bin`
`external/npm/node_modules/pkg_with_bin` -> `node_modules/pkg_with_bin`

Reply:
Oh. I see. This is so we can match the special case of
```
    if parts[0] == "external":
        if parts[2] == "node_modules":
            # external/npm/node_modules -> node_modules/foo
            # the linker will make sure we can resolve node_modules from npm
            return "/".join(parts[2:])
``` added 2 years ago.

That bit of logic seems wrong now that we can link into sub-directories... but also seems out of scope to fix that here

Reply:
Sorted it out. We don't want to strip the output path _unless_ we're transforming from `external/<npm>/node_modules/foo` to `node_modules/foo`. With that in mind, the function becomes
```
def _to_execroot_path(ctx, file):
    # Check for special case of external/<npm>/node_modules or
    # bazel-out/<platform>/bin/external/<npm>/node_modules.
    # TODO: This assumes we are linking this to the root node_modules which is not always the case
    # since the linker was updated to support linking to sub-directories since this special case was
    # added
    parts = file.path.split("/")
    if file.is_directory and not file.is_source:
        # Strip the output root to handle the case of a TreeArtifact exported by js_library with
        # exports directories only. Since a TreeArtifact is generated by an action, it resides
        # inside bazel-out directory.
        parts = parts[3:]
    if len(parts) > 3 and parts[0] == "external" and parts[2] == "node_modules":
        # Transform external/npm/node_modules/foo to node_modules/foo.
        # The linker will make sure we can resolve node_modules from npm.
        return "/".join(parts[2:])

    return file.path
```

Reply:
Other than that, this looks great. Thanks for tracking down the underlying issue @thesayyn . I'll push this little fix and this is landable.

Reply:
Ah great. It's more robust right now. Thanks!